braindecode/models/conformer.py

```diff
@@ -0,0 +1,279 @@
+
+        self.shallownet = nn.Sequential(
+            nn.Conv2d(1, kernel, (1, kernel_temp_conv), (1, 1)),
+            nn.Conv2d(kernel, kernel, (eeg_channel, 1), (1, 1)),
+            nn.BatchNorm2d(kernel),
+            nn.ELU(),
+            nn.AvgPool2d(
+                (1, kernel_avg_pool), (1, stride_avg_pool)
+            ),
+            # pooling acts as slicing to obtain 'patch' along the
+            # time dimension as in ViT
+            nn.Dropout(conv_drop),
+        )
+
+        self.projection = nn.Sequential(
+            nn.Conv2d(
+                kernel, kernel, (1, 1), stride=(1, 1)
+            ),  # transpose, conv could enhance fiting ability slightly
+            Rearrange("b e (h) (w) -> b (h w) e"),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = x.unsqueeze(dim=1)  # add one extra dimension
+        x = self.shallownet(x)
+        x = self.projection(x)
+        return x
+
+
+class MultiHeadAttention(nn.Module):
+    def __init__(self, emb_size, num_heads, dropout):
+        super().__init__()
+        self.emb_size = emb_size
+        self.num_heads = num_heads
+        self.keys = nn.Linear(emb_size, emb_size)
+        self.queries = nn.Linear(emb_size, emb_size)
+        self.values = nn.Linear(emb_size, emb_size)
+        self.att_drop = nn.Dropout(dropout)
+        self.projection = nn.Linear(emb_size, emb_size)
+
+    def forward(self, x: Tensor, mask: Tensor = None) -> Tensor:
+        queries = rearrange(
+            self.queries(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        keys = rearrange(
+            self.keys(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        values = rearrange(
+            self.values(x), "b n (h d) -> b h n d", h=self.num_heads
+        )
+        energy = torch.einsum("bhqd, bhkd -> bhqk", queries, keys)
+        if mask is not None:
+            fill_value = torch.finfo(torch.float32).min
+            energy.mask_fill(~mask, fill_value)
+
+        scaling = self.emb_size ** (1 / 2)
+        att = F.softmax(energy / scaling, dim=-1)
+        att = self.att_drop(att)
+        out = torch.einsum("bhal, bhlv -> bhav ", att, values)
+        out = rearrange(out, "b h n d -> b n (h d)")
+        out = self.projection(out)
+        return out
+
+
+class ResidualAdd(nn.Module):
+    def __init__(self, fn):
+        super().__init__()
+        self.fn = fn
+
+    def forward(self, x, **kwargs):
+        res = x
+        x = self.fn(x, **kwargs)
+        x += res
+        return x
+
+
+class FeedForwardBlock(nn.Sequential):
+    def __init__(self, emb_size, expansion, drop_p):
+        super().__init__(
+            nn.Linear(emb_size, expansion * emb_size),
+            nn.GELU(),
+            nn.Dropout(drop_p),
+            nn.Linear(expansion * emb_size, emb_size),
+        )
+
+
+class GELU(nn.Module):
+    def forward(self, input: Tensor) -> Tensor:
+        return input * 0.5 * (1.0 + torch.erf(input / math.sqrt(2.0)))
```

Comment:
use [nn.GELU](https://pytorch.org/docs/stable/generated/torch.nn.GELU.html) instead

Reply:
It's okay. Thank you!